### PR TITLE
Fix inference of instance attributes defined in base class

### LINKED ISSUE #932: Instance attributes defined in parent classes can raise InferenceError
### Steps to reproduce

Inferring an attribute on a class instance where the instance attribute is defined in a base class will fail, but
inferring that same attribute via `igetattr` on an `astroid.Instance` will succeed.

```python
import astroid
attr = astroid.extract_node("""
class Parent:
    def __init__(self):
        self.data = 123

class Test(Parent):
    pass
t = Test()
t.data
""")
# next(attr.infer())
# raises astroid.exceptions.InferenceError: Inference failed for <Attribute.data l.9 at 0x...>.
instance = next(attr.expr.infer())
print(next(instance.igetattr('data')))
# Const.int(value=123)
```

Discovered while debugging #904. I couldn't find an exact match for this issue before, but probably https://github.com/PyCQA/pylint/issues/960 is the same issue.

### Current behavior

As above

### Expected behavior

Inferring attribute should return `Const.int(value=123)`

### ``python -c "from astroid import __pkginfo__; print(__pkginfo__.version)"`` output

2.6.0-dev0

### PR DESCRIPTION
**Pull request by nelfin**
## Steps

- [x] For new features or bug fixes, add a ChangeLog entry describing what your PR does.
- [x] Write a good description on what the PR does.

## Description

Ref #932. The cause of this appears to be failing to specify the
`lookupname` on the inference context in `BaseInstance.igetattr`.
Additionally, the context path key was not set in a manner that matched
other usages. This appears to date back to at least 2015 when context
path keys were changed. The usage as a tuple instead of a node with
implict `lookupname` was changed last in 3d342e85 but the lookup key was
preserved throughout various changes since then.

## Type of Changes

|   | Type |
| ------------- | ------------- |
| ✓  | :bug: Bug fix  |

## Related Issue

Closes #932

### CODE REVIEW

**Comment by cdce8p** [comment]
@nelfin Would you mind resolving the merge conflicts? This PR seems like one I can review within the next few days.

File: tests/unittest_inference.py [hunk 1]
```diff
@@ -4050,6 +4050,42 @@ def inner():
...
+                self.data = 123
+
+        class Test(Parent):
+            pass
+        t = Test()
+        t
+        """
+        )
+        inferred = next(node.infer())
+        # self.assertIsInstance(inferred, bases.Instance)
```
Comment (cdce8p) [inline tests/unittest_inference.py:4067]: Why comment it out? Either test or remove it. The test should be fine

File: tests/unittest_inference.py [hunk 2]
```diff
@@ -4050,6 +4050,42 @@ def inner():
...
+        const = next(inferred.igetattr("data"))
+        self.assertIsInstance(const, nodes.Const)
+        self.assertEqual(const.value, 123)
```
Comment (cdce8p) [inline tests/unittest_inference.py:4070]: Can you use `assert isinstance` and `assert .. == ..` instead? We try to move more towards `pytest` and away from `unittest`. After you changed it, you probably need to add `@staticmethod` as well.

Same for the test case below.

**Review by cdce8p — CHANGES_REQUESTED** [review]
The change itself LGTM. Just some minor comments regarding the tests.

File: tests/unittest_inference.py (hunk 2 above)
Comment (nelfin) [reply tests/unittest_inference.py:4070]: Sure, I wasn't sure what the prevailing style was at the time so I just tried to match the surrounding tests

File: tests/unittest_inference.py (hunk 1 above)
Comment (nelfin) [reply tests/unittest_inference.py:4067]: Just cruft I overlooked

**Review by cdce8p — APPROVED** [review]
Thanks 🐬

**Comment by nelfin** [comment]
I know there's no perfect way to manage pull requests on GitHub, but I don't like how it disregarded all of the commit message and instead tells people to refer to this PR link. I'd rather that those messages stayed in the history.

**Comment by cdce8p** [comment]
> I know there's no perfect way to manage pull requests on GitHub, but I don't like how it disregarded all of the commit message and instead tells people to refer to this PR link. I'd rather that those messages stayed in the history.

Not completely sure about that. Up until now, when I merged something I decided on the go what made sense to keep. For this one, as it's just a small bugfix, I chose to remove it. I almost never look at a commit message outside of Github history where the PR is also available so I'm not the best one to judge. If you like, I can try to keep yours if I happen to merge any other of your PRs.

**Comment by nelfin** [comment]
> Not completely sure about that. Up until now, when I merged something I decided on the go what made sense to keep. For this one, as it's just a small bugfix, I chose to remove it.

That's fine, it's not likely to cause too much confusion. To me, the real value in a small bugfix is not the lines itself, but the explanation as to why things broke.

> I almost never look at a commit message outside of Github history where the PR is also available so I'm not the best one to judge.

This repository is actually older than GitHub itself so it's probably prudent to keep a "backup".

> If you like, I can try to keep yours if I happen to merge any other of your PRs.

If it's not too much trouble.

**Pull request merged**